nodejs/eks/nodegroup.ts

```diff
@@ -626,3 +624,83 @@ function tagsToAsgTags(tagsInput: InputTags): pulumi.Output<string> {
         return output;
     });
 }
+
+/**
+ * ManagedNodeGroupOptions describes the configuration options accepted by an
+ * EKS Managed NodeGroup.
+ *
+ * See for more details:
+ * https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html
+ */
+type ManagedNodeGroupOptions = Omit<aws.eks.NodeGroupArgs, "clusterName" | "subnetIds" | "scalingConfig"> & {
+    /**
+     * The target EKS cluster.
+     */
+    cluster: Cluster | CoreData;
+
+    /**
+     * Make clusterName optional, since the cluster is required and it contains it.
+     */
+    clusterName?: pulumi.Output<string>;
+
+    /**
+     * Make subnetIds optional, since the cluster is required and it contains it.
+     *
+     * Default subnetIds is chosen from the following list, in order, if
+     * subnetIds arg is not set:
+     *   - core.subnetIds
+     *   - core.privateIds
+     *   - core.publicSublicSubnetIds
+     */
+    subnetIds?: pulumi.Output<pulumi.Output<string>[]>;
+
+    /**
+     * Make scalingConfig optional, since defaults can be computed.
+     *
+     * Default scaling amounts of the node group autoscaling group are:
+     *   - desiredSize: 2
+     *   - minSize: 1
+     *   - maxSize: 2
+     */
+    scalingConfig?: pulumi.Input<awsInputs.eks.NodeGroupScalingConfig>
+};
+
+export function createManagedNodeGroup(name: string, args: ManagedNodeGroupOptions): aws.eks.NodeGroup {
+    const core = isCoreData(args.cluster) ? args.cluster : args.cluster.core;
+
+    // Compute the node group subnets to use.
+    let subnetIds: pulumi.Output<string[]> = pulumi.output([]);
+    if (args.subnetIds !== undefined) {
+        subnetIds = pulumi.output(args.subnetIds);
+    } else if (core.subnetIds !== undefined) {
+        subnetIds = core.subnetIds;
+    } else if (core.privateSubnetIds !== undefined) {
+        subnetIds = core.privateSubnetIds;
+    } else if (core.publicSubnetIds !== undefined) {
+        subnetIds = core.publicSubnetIds;
+    }
```

Comment:
If none of the above apply - should that be an error instead of falling back to `[]`?

Reply:
Also - I can't quite reason about whether the above is "right" - is this based on some precedent documented elsewhere either in AWS or in another part of our EKS package?

Reply:
> If none of the above apply - should that be an error instead of falling back to []?

Good catch. I'll fix

> Also - I can't quite reason about whether the above is "right" - is this based on some precedent documented elsewhere either in AWS or in another part of our EKS package?

This is based on our existing network config setup for self-managed node groups, and using sane defaults OOTB from the cluster object if the user does not specify `subnetIds`.

https://github.com/pulumi/pulumi-eks/blob/79fb423207e0906de815f0d08857f37b85bb14e7/nodejs/eks/nodegroup.ts#L458-L468

Do you think we should not be doing any sort of default logic?

Reply:
> Do you think we should not be doing any sort of default logic?

I think we should have default logic - I just wasn’t exactly sure how to know whether this specific logic was “right”. If we can cross reference any other source of truth for this - or even just document why we do what we do - that would help future maintainability.

Reply:
I spell this out in the comments:

https://github.com/pulumi/pulumi-eks/blob/41fc94a183c84d64fe0d0ebb4da368dab2d96bd7/nodejs/eks/nodegroup.ts#L646-L655

>  If we can cross reference any other source of truth for this - or even just document why we do what we do - that would help future maintainability.

This is just repeating the default logic, so I'll go ahead and add a note in the comments about reusing the `computeWorkerSubnet` approach here.